packages/build-tools/src/utils/expoUpdates.ts

```diff
@@ -88,14 +93,38 @@ export async function configureExpoUpdatesIfInstalledAsync(
   }
 
   const appConfigRuntimeVersion = ctx.job.version?.runtimeVersion ?? resolvedRuntimeVersion;
+
   if (ctx.metadata?.runtimeVersion && ctx.metadata.runtimeVersion !== appConfigRuntimeVersion) {
     ctx.markBuildPhaseHasWarnings();
-    ctx.logger.warn(
-      `Runtime version from the app config evaluated on your local machine (${ctx.metadata.runtimeVersion}) does not match the one resolved here (${appConfigRuntimeVersion}).`
-    );
-    ctx.logger.warn(
-      "If you're using conditional app configs, e.g. depending on an environment variable, make sure to set the variable in eas.json or configure it with EAS Secret."
-    );
+    ctx.logger.warn('Runtime version mismatch');
+    ctx.logger.warn(`Runtime version on your local machine: ${ctx.metadata.runtimeVersion}`);
+    ctx.logger.warn(`Runtime version calculated on EAS: ${appConfigRuntimeVersion}`);
+
+    if (
+      ctx.metadata?.projectFingerprintUrl &&
+      resolvedFingerprintSources &&
+      resolvedRuntimeVersion
+    ) {
+      try {
+        const result = await fetch(ctx.metadata?.projectFingerprintUrl);
```

Comment:
Ah, I see, this is where the URL is used from the build job.

@sjchmiela and @szdziedzic - What's the best way to access a file from within a GCS bucket in build tools is we only have the bucket key? Is it okay to put temporary data in the build job object like this expiring URL? Or is there a way to call www from build-tools to generate a fresh signed URL from a key?

Maybe this is kinda like the `secrets.robotAccessToken` which is also temporary data within the build job?

@kadikraman - if we do decide to put the temporary URL into the build job we'll need to increase how long the URL is good for to longer than the max queue time plus max build time to ensure the URL can still be fetched even in the worst case.

Reply:
This is now done in the turtle PR.